include/zephyr/drivers/flash.h

```diff
@@ -422,6 +425,60 @@ static inline const struct flash_parameters *z_impl_flash_get_parameters(const s
 	return api->get_parameters(dev);
 }
 
+/**
+ *  @brief Execute flash extended operation on given device
+ *
+ *  Besides of standard flash operations like write or erase, flash controllers
+ *  also support additional features like write protection or readout
+ *  protection. These features are not available in every flash controller,
+ *  what's more controllers can implement it in a different way.
+ *
+ *  It doesn't make sense to add a separate flash API function for every flash
+ *  controller feature, because it could be unique (supported on small number of
+ *  flash controllers) or the API won't be able to represent the same feature on
+ *  every flash controller.
+ *
+ *  Extended operation interface provides flexible way for supporting flash
+ *  controller features. Code space is divided equally into Zephyr codes
+ *  (MSb == 0) and vendor codes (MSb == 1). This way we can easily add extended
+ *  operations to the drivers without cluttering the API or problems with API
```

Comment:
With this PR, there should be some `FLASH_EX_OP_VENDOR_BASE` #define provided in flash.h, which will mark where the vendor space starts. This comment should be rather part of the `FLASH_EX_OP_VENDOR_BASE` definition.
It would be also nice if macro like `FLASH_EX_OP_IS_VENDOR()` macro would be provided that would allow to quickly find in which op space the code is.